### PR TITLE
this is the initial, running version of the scripts that we drop with the API zip

### PR DESCRIPTION
**Pull request by FreeAssoc**
it is really just kicking the tires and making sure that each of the scripts we've already given to Qumulo users doesn't break when using Qumulo REST API via pip.  If we don't want to drop these in public as-is, happy to adjust.

TODO: At least one top-level smoke test for each of these to ensure they haven't broken build over build.

### CODE REVIEW

**Comment by FreeAssoc** [comment]
the only script that is not included here is the `capture_config.py` script; I discussed this one with @dpirone-qumulo and we agreed that this particular script should not be included right now...

File: README.md [hunk 1]
```diff
@@ -19,7 +21,7 @@ the Qumulo REST Libarary -- you can read about it [here](https://pip.pypa.io/en/
 
 You'll need to install the Qumulo REST API in order to run any of these samples; You install it this way:
 
-pip install -r requirements.txt
```
Comment (dpirone-qumulo) [inline README.md]: @FreeAssoc - Until we make this "wrapper" public, this seems like a good place to mention how to get an access token.

File: directory_add.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+=== Options:
+
+[-u | --user] username              Use 'username' for authentication
+                                        (defaults to 'admin')
+[-p | --passwd] password            Use 'password' for authentication
+                                        (defaults to 'admin')
+[-P | --port] number                Use 'number' for the API server port
+                                        (defaults to 8000)
+
+[-V | --var] demo_var               Use 'demo_var' as the demo-specific
```
Comment (dpirone-qumulo) [inline directory_add.py]: Please say what `demo_var` is in each of the scripts.

File: directory_add.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+sys.path.append(os.path.join(os.path.dirname(__file__), ".."))
+import qumulo.lib.auth
+import qumulo.lib.request
+import qumulo.rest
+
+#### Classes
+class AddDirectoryCommand(object):
+    ''' class wrapper for REST API command so that we can new them up in tests '''
+    def __init__(self, argv=None):
+        self.host = '127.0.0.1'
```
Comment (dpirone-qumulo) [inline directory_add.py]: I think a better option for the qumulo cluster host name is to NOT default it, and instead force the user to specify it on the command line.

**Comment by dpirone-qumulo** [comment]
Looking great!  I would encourage you to run it through pylint just to make sure we don't have any lingering cruft.

File: tests/test_scripts.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+from group_delete import DeleteGroupCommand
+from user_add import AddUserCommand
+from user_delete import DeleteUserCommand
+from directory_add import AddDirectoryCommand
+from directory_delete import DeleteDirectoryCommand
+from nfs_add_share import AddNFSShareCommand
+from nfs_delete_share import DeleteNFSShareCommand
+
+
+class TestClass(unittest.TestCase):
```
Comment (dpirone-qumulo) [inline tests/test_scripts.py]: Give this a more descriptive class Name

File: tests/test_scripts.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+    def setUp(self):
+        # tweak these to match your local environment
+        self.test_argv = ['./auth_add_group_request.py',
+                           '--host',
+                           'dev',
+                           '--port',
+                           '23831',
```
Comment (dpirone-qumulo) [inline tests/test_scripts.py]: Change to well know port ( 8000 )

File: version_request.py [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+                self.passwd = arg
+            elif opt in ("--var", "-V"):
+                self.demo_var = arg
+            else:
+                print "Unrecognized option %s\n" % opt
+                print __doc__
+                sys.exit(1)
+
+#### Subroutines
+def login(host, user, passwd, port):
```
Comment (dpirone-qumulo) [inline version_request.py]: Since we don't need to login to do the version api call - I think we can stream line this code.

File: README.md (hunk 1 above)
Comment (FreeAssoc) [inline README.md]: added, thanks

File: directory_add.py (hunk 2 above)
Comment (FreeAssoc) [inline directory_add.py]: fixed - thx

File: tests/test_scripts.py (hunk 4 above)
Comment (FreeAssoc) [inline tests/test_scripts.py]: fixed (changed to `CommandTests`)

File: tests/test_scripts.py (hunk 5 above)
Comment (FreeAssoc) [inline tests/test_scripts.py]: this makes the tests pass for my local environment -- has a comment that mentions changing the port for local use

File: version_request.py (hunk 6 above)
Comment (FreeAssoc) [inline version_request.py]: removed login method from version_request - thanks

**Pull request merged**